### ISSUE TITLE
Why do I train my dataset and LR appears 0？

### ISSUE THREAD

**Issue by Kingxudong** [facebookresearch/dino#196]
Will run the code on one GPU.
| distributed init (rank 0): env://
fatal: Not a git repository (or any parent up to mount point /home1)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
git:
  sha: N/A, status: clean, branch: N/A

arch: vit_small
batch_size_per_gpu: 4
clip_grad: 3.0
data_path: /home1/wangxudong/dino/data/train/
dist_url: env://
drop_path_rate: 0.1
epochs: 100
freeze_last_layer: 1
global_crops_scale: (0.4, 1.0)
gpu: 0
local_crops_number: 8
local_crops_scale: (0.05, 0.4)
local_rank: 0
lr: 0.003
min_lr: 1e-06
momentum_teacher: 0.996
norm_last_layer: True
num_workers: 10
optimizer: adamw
out_dim: 65536
output_dir: /home1/wangxudong/dino/result/
patch_size: 8
rank: 0
saveckp_freq: 20
seed: 0
teacher_temp: 0.04
use_bn_in_head: False
use_fp16: True
warmup_epochs: 10
warmup_teacher_temp: 0.04
warmup_teacher_temp_epochs: 0
weight_decay: 0.04
weight_decay_end: 0.4
world_size: 1
Data loaded: there are 57189 images.
Student and Teacher are built: they are both vit_small network.
Loss, optimizer and schedulers ready.
Starting DINO training !
/home/hanwei-5/anaconda3/envs/CenterNet-MIAOD/lib/python3.6/site-packages/torch/nn/functional.py:3121: UserWarning: Default upsampling behavior when mode=bicubic is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
/home/hanwei-5/anaconda3/envs/CenterNet-MIAOD/lib/python3.6/site-packages/torch/nn/functional.py:3000: UserWarning: The default behavior for interpolate/upsample with float scale_factor changed in 1.6.0 to align with other frameworks/libraries, and uses scale_factor directly, instead of relying on the computed output size. If you wish to keep the old behavior, please set recompute_scale_factor=True. See the documentation of nn.Upsample for details. 
  warnings.warn("The default behavior for interpolate/upsample with float scale_factor changed "
Epoch: [0/100]  [    0/14297]  eta: 6:50:32  loss: 10.387223 (10.387223)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 1.722933  data: 0.781645  max mem: 6112
Epoch: [0/100]  [   10/14297]  eta: 1:44:33  loss: 10.750419 (10.716266)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.439104  data: 0.071124  max mem: 6442
Epoch: [0/100]  [   20/14297]  eta: 1:29:40  loss: 10.956452 (10.891556)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.309576  data: 0.000065  max mem: 6442
Epoch: [0/100]  [   30/14297]  eta: 1:24:30  loss: 11.138003 (10.984449)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.309372  data: 0.000064  max mem: 6442
Epoch: [0/100]  [   40/14297]  eta: 1:21:58  loss: 11.210602 (11.043648)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.311484  data: 0.000078  max mem: 6442
Epoch: [0/100]  [   50/14297]  eta: 1:20:29  loss: 11.240683 (11.085256)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.313553  data: 0.000085  max mem: 6442
Epoch: [0/100]  [   60/14297]  eta: 1:19:22  loss: 11.252364 (11.112735)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.313072  data: 0.000077  max mem: 6442
Epoch: [0/100]  [   70/14297]  eta: 1:18:33  loss: 11.254496 (11.131395)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.311629  data: 0.000068  max mem: 6442
Epoch: [0/100]  [   80/14297]  eta: 1:17:55  loss: 11.257144 (11.147026)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.311581  data: 0.000067  max mem: 6442
Epoch: [0/100]  [   90/14297]  eta: 1:17:28  loss: 11.262831 (11.159023)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.312679  data: 0.000079  max mem: 6442
Epoch: [0/100]  [  100/14297]  eta: 1:17:04  loss: 11.262872 (11.169433)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.313174  data: 0.000077  max mem: 6442
Epoch: [0/100]  [  110/14297]  eta: 1:16:44  loss: 11.252973 (11.176128)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.312549  data: 0.000070  max mem: 6442
Epoch: [0/100]  [  120/14297]  eta: 1:16:25  loss: 11.246675 (11.181134)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.311828  data: 0.000067  max mem: 6442
Epoch: [0/100]  [  130/14297]  eta: 1:16:12  loss: 11.247989 (11.185959)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.312724  data: 0.000072  max mem: 6442
Epoch: [0/100]  [  140/14297]  eta: 1:16:01  loss: 11.252209 (11.190857)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314582  data: 0.000080  max mem: 6442
Epoch: [0/100]  [  150/14297]  eta: 1:15:50  loss: 11.255443 (11.195298)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314306  data: 0.000074  max mem: 6442
Epoch: [0/100]  [  160/14297]  eta: 1:15:46  loss: 11.255443 (11.198935)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.317462  data: 0.000076  max mem: 6442
Epoch: [0/100]  [  170/14297]  eta: 1:15:44  loss: 11.255864 (11.202338)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.321913  data: 0.000077  max mem: 6442
Epoch: [0/100]  [  180/14297]  eta: 1:15:56  loss: 11.257596 (11.205689)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.332308  data: 0.000077  max mem: 6442
Epoch: [0/100]  [  190/14297]  eta: 1:15:55  loss: 11.264005 (11.208919)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.333600  data: 0.000077  max mem: 6442
Epoch: [0/100]  [  200/14297]  eta: 1:15:46  loss: 11.261575 (11.211577)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.319800  data: 0.000075  max mem: 6442
Epoch: [0/100]  [  210/14297]  eta: 1:15:37  loss: 11.253565 (11.213242)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314444  data: 0.000077  max mem: 6442
Epoch: [0/100]  [  220/14297]  eta: 1:15:29  loss: 11.252179 (11.215122)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314212  data: 0.000075  max mem: 6442
Epoch: [0/100]  [  230/14297]  eta: 1:15:22  loss: 11.253459 (11.216818)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314606  data: 0.000073  max mem: 6442
Epoch: [0/100]  [  240/14297]  eta: 1:15:14  loss: 11.260520 (11.218827)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314539  data: 0.000068  max mem: 6442
Epoch: [0/100]  [  250/14297]  eta: 1:15:07  loss: 11.260520 (11.220317)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.314548  data: 0.000069  max mem: 6442
Epoch: [0/100]  [  260/14297]  eta: 1:15:01  loss: 11.257047 (11.221865)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.315426  data: 0.000071  max mem: 6442
Epoch: [0/100]  [  270/14297]  eta: 1:14:55  loss: 11.260763 (11.223471)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.315098  data: 0.000068  max mem: 6442
Epoch: [0/100]  [  280/14297]  eta: 1:14:51  loss: 11.257568 (11.224644)  lr: 0.000000 (0.000000)  wd: 0.040000 (0.040000)  time: 0.316770  data: 0.000070  max mem: 6442

**Comment by Harry-KIT** [comment]
Hi @Kingxudong. You need a huge dataset. Because ViT requires a lot of datapoints for training.
In addition, fine-tuning can help to boost performance. Please check #80
